### ISSUE TITLE
Downtime lost after restart

### ISSUE THREAD

**Issue by cite** [Icinga/icinga2#5625]
After restarting Icinga 2, downtimes for services are lost.

JSON output (`/v1/objects/services?service=muc1pro-ite-1!child-health`) for service `muc1pro-ite-1!child-health` before restart:

```
{
  "results": [
    {
      "attrs": {
        "__name": "muc1pro-ite-1!child-health",
[...]
        "display_name": "child-health",
       "downtime_depth": 1,
[...]
```

And after restarting (even WITHOUT any configuration change):

```
{
  "results": [
    {
      "attrs": {
        "__name": "muc1pro-ite-1!child-health",
    [...]
        "display_name": "child-health",
        "downtime_depth": 0,
```

What other data would you need me to provide?

## Your Environment

* Version used (`icinga2 --version`): happens with stock 2.7.1 as well as a 2.7.1 where we included https://github.com/Icinga/icinga2/commit/ef5013b9038b07afc228122bade44dac52396a46 and https://github.com/Icinga/icinga2/commit/1cb39994a565c59df0ebffd424e80e5f898a4181
* Operating System and version:
CentOS 6.9, 64bit
* Enabled features (`icinga2 feature list`):
debuglog gelf graphite livestatus opentsdb perfdata syslog
* Config validation (`icinga2 daemon -C`):
```
information/cli: Icinga application loader (version: r2.7.1-1)
information/cli: Loading configuration file(s).
information/ConfigItem: Committing config item(s).
information/ApiListener: My API identity: fra1pro-infra-master-1.example.com
warning/InfluxdbWriter: 'socket_timeout' option has no effect and will be removed in Icinga 2 v2.8
warning/ApplyRule: Apply rule 'CHART-SI' (in /etc/icinga2/zones.d/global-configuration/services/active/chartserver.conf: 20:1-20:24) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'F5_SYNC' (in /etc/icinga2/zones.d/global-configuration/services/active/f5_sync.conf: 7:1-7:23) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'FDELAYD' (in /etc/icinga2/zones.d/global-configuration/services/active/fdelayd.conf: 7:1-7:23) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'FRESHCLAM' (in /etc/icinga2/zones.d/global-configuration/services/active/freshclam.conf: 7:1-7:25) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'HTTP_PDF' (in /etc/icinga2/zones.d/global-configuration/services/active/http_pdf.conf: 7:1-7:24) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'HYPERVISOR_DF_SUSPED' (in /etc/icinga2/zones.d/global-configuration/services/active/hypervisor.conf: 7:1-7:36) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'JBOSS' (in /etc/icinga2/zones.d/global-configuration/services/active/jboss.conf: 7:1-7:21) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'NFS_MOUNTPOINT ' (in /etc/icinga2/zones.d/global-configuration/services/active/nfs_mountpoints.conf: 8:1-8:81) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'p2ps' (in /etc/icinga2/zones.d/global-configuration/services/active/p2ps.conf: 7:1-7:20) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'RRCPD' (in /etc/icinga2/zones.d/global-configuration/services/active/rrcpd.conf: 7:1-7:21) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'SRC_DIST' (in /etc/icinga2/zones.d/global-configuration/services/active/src_dist.conf: 7:1-7:24) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'STATISTIC_SI' (in /etc/icinga2/zones.d/global-configuration/services/active/statistic_si.conf: 7:1-7:28) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'TOMCAT_JMX' (in /etc/icinga2/zones.d/global-configuration/services/active/tomcat_jmx.conf: 7:1-7:26) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'NRPE_FRS' (in /etc/icinga2/zones.d/global-configuration/services/active/windows_frs.conf: 7:1-7:24) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'fcom_simcorp_proxy_receiver' (in /etc/icinga2/zones.d/global-configuration/services/passive/fcom_simcorp_proxy_receiver.conf: 7:1-7:43) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'fidswapper_conflation_rt' (in /etc/icinga2/zones.d/global-configuration/services/passive/fidswapper_conflation_rt.conf: 7:1-7:40) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'fidswapper_conflation_rt_check_logs' (in /etc/icinga2/zones.d/global-configuration/services/passive/fidswapper_conflation_rt.conf: 13:1-13:51) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'ldap_repl_conn' (in /etc/icinga2/zones.d/global-configuration/services/passive/ldap_repl_conn.conf: 7:1-7:30) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'mailq' (in /etc/icinga2/zones.d/global-configuration/services/passive/mailq.conf: 10:1-10:21) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'mysql_ndb_cluster_node' (in /etc/icinga2/zones.d/global-configuration/services/passive/mysql_ndb_cluster_node.conf: 7:1-7:38) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'scoach_adapter' (in /etc/icinga2/zones.d/global-configuration/services/passive/scoach_adapter.conf: 7:1-7:30) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'scoach_adapter_check_logs' (in /etc/icinga2/zones.d/global-configuration/services/passive/scoach_adapter.conf: 13:1-13:41) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'servicecheck_bb' (in /etc/icinga2/zones.d/global-configuration/services/passive/servicecheck_bb.conf: 7:1-7:31) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'servicecheck_brokerstats_feedcache' (in /etc/icinga2/zones.d/global-configuration/services/passive/servicecheck_brokerstats.conf: 7:1-7:50) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'servicecheck_fe' (in /etc/icinga2/zones.d/global-configuration/services/passive/servicecheck_fe.conf: 7:1-7:31) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'streamingsvc' (in /etc/icinga2/zones.d/global-configuration/services/passive/streamingsvc.conf: 7:1-7:28) for type 'Service' does not match anywhere!
warning/ApplyRule: Apply rule 'streamingsvc_check_logs' (in /etc/icinga2/zones.d/global-configuration/services/passive/streamingsvc.conf: 13:1-13:39) for type 'Service' does not match anywhere!
information/ConfigItem: Instantiated 5 ApiUsers.
information/ConfigItem: Instantiated 1 ApiListener.
information/ConfigItem: Instantiated 588 Zones.
information/ConfigItem: Instantiated 1 FileLogger.
information/ConfigItem: Instantiated 599 Endpoints.
information/ConfigItem: Instantiated 9 UserGroups.
information/ConfigItem: Instantiated 34478 Notifications.
information/ConfigItem: Instantiated 6 NotificationCommands.
information/ConfigItem: Instantiated 252 CheckCommands.
information/ConfigItem: Instantiated 14 HostGroups.
information/ConfigItem: Instantiated 1 IcingaApplication.
information/ConfigItem: Instantiated 895 Hosts.
information/ConfigItem: Instantiated 8112 Dependencies.
information/ConfigItem: Instantiated 24 Users.
information/ConfigItem: Instantiated 5 TimePeriods.
information/ConfigItem: Instantiated 9895 Services.
information/ConfigItem: Instantiated 1 CompatLogger.
information/ConfigItem: Instantiated 1 StatusDataWriter.
information/ConfigItem: Instantiated 1 ExternalCommandListener.
information/ConfigItem: Instantiated 1 CheckerComponent.
information/ConfigItem: Instantiated 1 IdoMysqlConnection.
information/ConfigItem: Instantiated 1 InfluxdbWriter.
information/ConfigItem: Instantiated 1 NotificationComponent.
information/ScriptGlobal: Dumping variables to file '/var/cache/icinga2/icinga2.vars'
information/cli: Finished validating the configuration file(s).
```

**Comment by dnsmichi** [comment]
Verify the package structure as mentioned in https://github.com/Icinga/icinga2/issues/3668#issuecomment-282549005 and post your findings here please.

**Comment by cite** [comment]
Performing the steps mentioned in your link fixed the problem for the primary configuration master -
 thanks a lot. Our installation was missing any files in `/var/lib/icinga2/api/packages/_api`, and also had an extraneous `conf.d` directory containing comments and downtimes. This is now fixed on the primary configuration master, and the downtimes are visible again.

What would be the easiest way to fix this across the other members of the master zone, our 22 satellites and all clients (it it needs fixing), and how do we prevent this from happening again?

**Comment by dnsmichi** [comment]
This should be addressed by #5620 which ensures that the activestage name is always set and a package creation is atomoc. In terms of fixing the package - deleting it on the secondary master/satellites should be sufficient. Or you'll manually rsync the stage content, if the sync takes too long.

**Comment by cite** [comment]
Ok, thank you for your help. Closing this issue.
